core/src/main/java/com/pholser/junit/quickcheck/internal/generator/GeneratorRepository.java

```diff
@@ -362,4 +374,16 @@ private boolean hasGeneratorsFor(ParameterTypeContext parameter) {
     private static org.javaruntype.type.Type<?> token(Type type) {
         return Types.forJavaLangReflectType(type);
     }
+
+    private static boolean isPrimitiveType(Type type) {
```

Comment:
You may be able to avoid compiler warnings about raw types by phrasing as:

    return type instanceof Class<?>  && ((Class<?>) type).isPrimitive();

You can use the straight-wildcard form of a generic type in `instanceof` and casts without issue.